docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -245,6 +245,29 @@ For flat typed networks, regular kube-proxy may be used to provide pod access to
 For VPC-isolated networking, service IP support has to be provided w/o relying on iptable/IPVS rules on host netns. In other words, traditional kube-proxy can not used in VPC-isolated networking. A dedicated controller may be employed to establish & maintain service IP/pod IP mappings.
 (**TBD: more details required**)
 
+### Endpoints
+Each endpoints object associates with a network, the same as its service.
+
+For the services of _kubernetes_ and _kube-dns_ (in fact they are kubernetes_\{network\} and kube-dns_\{network\}), there are two special set of endpoints objects:
+
+| endpoints of | example | notes |
+| ---|--- | --- |
+| kubernetes related | kubernetes_net1 | if in default ns, it is for the kubernetes service in network net1
+| kube-dns related | kube-dns_net2 | if in kube-system ns, it is for the kube-dns service in network net2
+
+In order to prevent potential name conflicts, production system is hardened to prohibits regular users to name endpoints as kubernetes_{string} in default ns, and kube-dns_{string} in kube-system ns. They shall be reserved for controllers only.
```

Comment:
From authentication and authorization perspective, there is no way to tell the difference between a human user and a controller.  It's just a string user name. Maybe just let user "system:xxxx" can create these endpoints?  Or simply embed a short random ID in the name of endpoints, like how replicaSet name the generated pods?

Reply:
will rewrite this section to clarify.

Reply:
actually I will delete this section to avoid unnecessary confusion. If the named EP has been created, it should be accepted behavior for system to report proper error so that user can engage accordingly. The original sentences intended to state that there is finer-grained access control we may be able to leverage to prevent such fault from existence at the first place. The finer-grained access control can be in form of custom admission control, or modified api-server  code.